src/partitioning/metis_partitioner.C

```diff
@@ -403,8 +403,16 @@ void MetisPartitioner::_do_partition (MeshBase & mesh,
                     // would be nice
                     Elem * neighbor = const_cast<Elem *>(*n_it);
 
-                    csr_graph(elem_global_index, connection++) =
-                      global_index_map[neighbor->id()];
+                    // Not all interior neighbors are necessarily in
+                    // the global_index_map.  This will be the case
+                    // when partitioning a BoundaryMesh, whose
+                    // elements all have interior_parents() that
+                    // belong to some other Mesh.
+                    vectormap<dof_id_type, dof_id_type>::iterator global_index_map_it =
+                      global_index_map.find(neighbor->id());
+
```

Comment:
This isn't a sufficient fix, though, is it?  The find will succeed if the current mesh has an element whose ID is the same as the interior_parent, whether or not that element *is* the interior_parent.  To actually be sure we've got the interior_parent on the same mesh we'd have to query the mesh for the element having that ID and make sure the pointers match.